test/e2e/gameserver_test.go

```diff
@@ -1147,3 +1147,50 @@ func TestPlayerConnectAndDisconnect(t *testing.T) {
 	assert.NoError(t, err)
 	assert.ElementsMatch(t, []string{"1", "3"}, gs.Status.Players.IDs)
 }
+
+func TestGracefulShutdown(t *testing.T) {
+	if !runtime.FeatureEnabled(runtime.FeatureSDKGracefulTermination) {
+		t.SkipNow()
+	}
+	t.Parallel()
+
+	log := e2eframework.TestLogger(t)
+	ctx := context.Background()
+	gs := framework.DefaultGameServer(framework.Namespace)
+	var minute int64 = 60
+	gs.Spec.Template.Spec.TerminationGracePeriodSeconds = &minute
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, framework.Namespace, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
+	gameservers := framework.AgonesClient.AgonesV1().GameServers(framework.Namespace)
+	err = gameservers.Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{})
+	require.NoError(t, err)
+	log.Info("Created GameServer, waited 20 seconds")
```

Comment:
This log line is right after you are calling Delete (and before the 20 second sleep), which is confusing. Should it be: "Deleted GameServer, waiting 20 seconds..."

Reply:
Yes it should - good catch. Updating.

Reply:
In the new code I still see `log.Info("Created GameServer, waiting 20 seconds...")`

Reply:
🤦🏻 I read the second part of the sentence, not the first.